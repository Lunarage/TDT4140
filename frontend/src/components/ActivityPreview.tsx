import React from 'react'
import styled from 'styled-components';
import { redHexColor, redHexColorHover } from '../consts';

const Wrapper = styled.div`
  background-color: ${redHexColor};
  width: 317px;
  height: 724px;
  box-shadow: 1px 1px 20px 4px rgba(0, 0, 0, 0.25);
  top: 354px;
  left: 400px;
  color: white;
  mix-blend-mode: normal;
  font-style: bold;
  font-weight: 500;
  font-size: 20px;
  line-height: 27px;

  &:hover {
    cursor: pointer;
    box-shadow: 1px 1px 20px 4px rgba(0, 0, 0, 0.5);
    background-color: ${redHexColorHover};
  }
`;

const TextWrapper = styled.div`
  padding: 20px;
`;

const Image = styled.img`
  width: 100%;
  height: 352px;
  box-shadow: 1px 1px 20px 4px rgba(0, 0, 0, 0.25);
`;

const ActivityPreview = () => (
  <Wrapper>
    <Image src="Red-Rocks_horizontal.jpg" /> 
    <TextWrapper>
    <h2> KLATRING </h2> 
    <div><b>Inne/Ute: </b> Utendørs</div><br></br>
    <div><b>Sted: </b>Lade </div>
    <div><b>Dato: </b> 12. Februar 2021 </div>
    <div><b>Klokkeslett: </b> 14.00 - 16.00 </div>
    <div><b>Aktivitetsnivå: </b> 3 </div>
    <div><b>Kategori: </b> Sport og friluftsliv </div>
    </TextWrapper>

  </Wrapper>
);

export default ActivityPreview
