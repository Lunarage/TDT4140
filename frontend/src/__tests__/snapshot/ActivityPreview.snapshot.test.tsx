import React from "react";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import ActivityPreview from '../../components/ActivityPreview';

Enzyme.configure({ adapter: new Adapter() });

describe("ActivityPreview", () => {
  it("renders ActivityPreview", () => {
    shallow(<ActivityPreview onClickFunc={() => { }} />);
  });
  it("Should match ActivityPreview snapshot", () => {
    const snap = shallow(<ActivityPreview onClickFunc={() => { }} />);
    expect(toJSON(snap)).toMatchSnapshot();
  });
});
