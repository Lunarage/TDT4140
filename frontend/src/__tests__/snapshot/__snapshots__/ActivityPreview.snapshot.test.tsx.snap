// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityPreview Should match ActivityPreview snapshot 1`] = `
<ActivityPreviewWrapper>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ActivityPreview
      data={
        Object {
          "activity_level": 1,
          "categories": Array [
            1,
          ],
          "categories_names": Array [
            "cat",
          ],
          "date": "2021-06-06T20:20:20Z",
          "description": "desc",
          "equipment_used": Array [
            1,
          ],
          "equipment_used_names": Array [
            "eq",
          ],
          "id": 1,
          "location": "loc",
          "max_participants": 1,
          "organization_owner": 1,
          "organization_owner_name": "org",
          "title": "test",
          "user_owner": 1,
          "user_owner_username": "user",
        }
      }
      onClickFunc={[Function]}
    >
      <styled.div
        onClick={[Function]}
      >
        <div
          className="sc-bdfBwQ lmgbPw"
          onClick={[Function]}
        >
          <styled.div>
            <div
              className="sc-jSgupP jKWuSU"
            >
              <styled.img
                src="static/Red-Rocks_horizontal.jpg"
              >
                <img
                  className="sc-dlfnbm cDnPeZ"
                  src="static/Red-Rocks_horizontal.jpg"
                />
              </styled.img>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="sc-gsTCUz foYBRd"
            >
              <styled.h2>
                <h2
                  className="sc-hKgILt dbvOfW"
                >
                   
                  test
                   
                </h2>
              </styled.h2>
              <styled.div>
                <div
                  className="sc-eCssSg kVXabg"
                >
                  <b>
                    Sted: 
                  </b>
                  loc
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-eCssSg kVXabg"
                >
                  <b>
                    Dato: 
                  </b>
                   
                  06
                  /
                  06
                   
                  2021
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-eCssSg kVXabg"
                >
                  <b>
                    Aktivitetsniv√•: 
                  </b>
                   
                  1
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-eCssSg kVXabg"
                >
                  <b>
                    Kategorier: 
                  </b>
                   
                  cat
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-eCssSg kVXabg"
                >
                  <b>
                    Organization: 
                  </b>
                   
                  org
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </ActivityPreview>
  </Provider>
</ActivityPreviewWrapper>
`;
